### ISSUE TITLE
build prefers prebuilt nuget packages over source-built nuget packages in some cases

### ISSUE THREAD

**Issue by tmds** [dotnet/source-build#4208]
While validating https://github.com/dotnet/source-build-externals/pull/281 I see the following (vmr main, --online build):

In the SDK that gets built `./sdk/9.0.100-preview.3.24161.1/DotnetTools/dotnet-user-jwts/9.0.0-preview.3.24158.2/tools/net9.0/any/Microsoft.IdentityModel.Tokens.dll` has `/vmr/src/source-build-externals/artifacts/sb/src/src/azure-activedirectory-identitymodel-extensions-for-dotnet/src/Microsoft.IdentityModel.Tokens/obj/Release/netstandard2.0/Microsoft.IdentityModel.Tokens.pdb` stored for the pdb path.

This path is where the prebuilt nuget package was built.
In other words: the build is consuming the prebuilt nuget package when building `dotnet-user-jwts`.

The source-built nuget package for `Microsoft.IdentityModel.Tokens.7.1.2.nupkg` has a mapped path (as intended by the PR): `/_/src/source-build-externals/artifacts/sb/src/src/azure-activedirectory-identitymodel-extensions-for-dotnet/src/Microsoft.IdentityModel.Tokens/obj/Release/netstandard2.0/Microsoft.IdentityModel.Tokens.pdb`.

cc @MichaelSimons @mthalman @omajid

**Comment by mthalman** [comment]
These two pdb paths look the same to me. What am I missing?

**Comment by tmds** [comment]
`/vmr/src/...` is where the prebuilt nuget package dll was built.

`/_/src/...` is where the source-build dll got path-mapped with `DeterministicSourcePaths` (https://github.com/dotnet/source-build-externals/pull/281).

**Comment by tmds** [comment]
@omajid pointed me to https://github.com/dotnet/source-build/issues/4206 which is probably the root cause of this issue. When https://github.com/dotnet/source-build/issues/4206 is fixed, I'll verify this issue is resolved and close it.

**Comment by mthalman** [comment]
> @omajid pointed me to #4206 which is probably the root cause of this issue. When #4206 is fixed, I'll verify this issue is resolved and close it.

That issue only applies when using the official SDK Preview 2 release for your toolset.

**Comment by tmds** [comment]
This is the version that my builds used:
```
Welcome to .NET 9.0!
---------------------
SDK Version: 9.0.100-preview.2.24154.2
```

**Comment by MichaelSimons** [comment]
Closing in preference of using https://github.com/dotnet/source-build/issues/4206 to track this problem.
